src/test/scala/tech/cryptonomic/conseil/tezos/DataTypesTest.scala

```diff
@@ -213,23 +243,52 @@ import scala.concurrent.ExecutionContext.Implicits.global
         aggregation = Some(List(ApiAggregation(field = "invalid")))
       )
 
-      val result = query.validate("test", tpdo)
+      val result = query.validate(testEntityPath, ms)
 
       result.futureValue.left.get should contain theSameElementsAs List(InvalidAggregationField("invalid"), InvalidAggregationFieldForType("invalid"))
     }
 
+    "return error with field XXXXX" in {
+      val attribute = Attribute(
+        name = "InvalidAttribute",
+        displayName = "Invalid",
+        dataType = DataType.String,
+        cardinality = None,
+        keyType = KeyType.NonKey,
+        entity = "test"
+      )
+
+      (ms.isAttributeValid _).when("testEntity", "InvalidAttribute").returns(Future.successful(true))
+      (ms.getTableAttributesWithoutUpdatingCache(_: EntityPath)(_: ExecutionContext)).when(*, *).returns(Future.successful(Some(List(attribute))))
+      (ms.getEntities(_: NetworkPath)(_: ExecutionContext)).when(*, *).returns(Future.successful(Some(List(testEntity.copy(limitedQuery = Some(true))))))
+
+      val query = ApiQuery(
+        fields = None,
+        predicates = Some(List(ApiPredicate("InvalidAttribute", OperationType.in))),
+        orderBy = None,
+        limit = None,
+        output = None,
+        aggregation = None
+      )
+
+      val result = query.validate(testEntityPath, ms)
+
+      result.futureValue.left.get shouldBe List(InvalidPredicateFiltering("None of attributes InvalidAttribute is valid for predicate"))
```

Comment:
This might be arguable, but I find that hardcoding the expected error message in tests might prove painful to maintain with future code changes. WDYT?

Reply:
can scalatest do the comparison somehow? or did you have something else in mind?

Reply:
`... shouldBe a[List[InvalidPredicateFiltering]]`

or something similar, i.e. only checking the error type, not the message

Reply:
I've got this kind of warning about type erasure:
`Type parameter should not be specified because it will be erased at runtime, please use _ instead.  Note that in future version of ScalaTest this will give a compiler error.`, I won't change it for now.

Reply:
yes, you can solve this by something equivalent like
```scala
val errors = result.futureValue.left.get
errors should have size 1
errors.head shouldBe a[InvalidPredicateFiltering]
```

Reply:
yes, done it